UN-OCHA/ocha_basic
# CD-23: Support PWA Drupal module

Similar to #1 the [Web Manifest](https://developer.mozilla.org/en-US/docs/Web/Manifest) in this theme is hardcoded, which doesn't play well with other modules in a Drupal implementation that also want to customize the manifest.

Additionally, the manifest.json is very unlikely to be used without a valid Service Worker, which this theme lacks. Supporting the [PWA Drupal module](https://www.drupal.org/project/pwa) is an easy way to achieve basic offline support and "Add to Homescreen" prompts without having to worry about authoring and maintaining a custom Service Worker.

There is no negative effect when the PWA module is disabled or not yet installed. It's a pure enhancement which does not affect basic website functionality.

### Testing

- Download/Enable PWA module
- Enable `ocha_basic` and set to default
- Clear all caches
- In Chrome DevTools, open the **Application** tab and in the left-hand column click **Manifest**
- Manifest should claim to live at `sites/default/files/pwa/manifest.json`
- Inspect the manifest metadata in the right-hand column and compare to reference screenshot:

<img width="997" alt="pwa-ocha_basic" src="https://user-images.githubusercontent.com/254753/42750458-9bbcea3a-88e7-11e8-9a3c-43c459962345.png">
